pkg/webhooks/subscription/subscription.go

```diff
@@ -0,0 +1,182 @@
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName    string = "subscription-validation"
+	LoggingSubName string = "cluster-logging"
+	ESSubName      string = "elasticsearch-operator"
+	Channel        string = "4.5"
+)
+
+var (
+	privilegedUsers = []string{"kube:admin", "system:admin", "system:serviceaccount:openshift-authentication:oauth-openshift"}
+	adminGroups     = []string{"osd-sre-admins", "osd-sre-cluster-admins"}
+
+	log = logf.Log.WithName(WebhookName)
+
+	scope = admissionregv1.ClusterScope
+	rules = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{"UPDATE", "CREATE"},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"operators.coreos.com"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"subscriptions"},
+				Scope:       &scope,
+			},
+		},
+	}
+)
+
+type subscriptionRequest struct {
+	Metadata struct {
+		Name      string `json:"name"`
+		Namespace string `json:"namespace"`
+	} `json:"metadata"`
+	Spec struct {
+		Channel string `json:"channel"`
+		Name    string `json:"name"`
+	} `json:"spec"`
+}
+
+// SubscriptionWebhook validates a Subscription change
+type SubscriptionWebhook struct {
+	mu sync.Mutex
+	s  runtime.Scheme
+}
+
+// TimeoutSeconds implements Webhook interface
+func (s *SubscriptionWebhook) TimeoutSeconds() int32 { return 2 }
+
+// MatchPolicy implements Webhook interface
+func (s *SubscriptionWebhook) MatchPolicy() admissionregv1.MatchPolicyType {
+	return admissionregv1.Equivalent
+}
+
+// Name implements Webhook interface
+func (s *SubscriptionWebhook) Name() string { return WebhookName }
+
+// FailurePolicy implements Webhook interface
+func (s *SubscriptionWebhook) FailurePolicy() admissionregv1.FailurePolicyType {
+	return admissionregv1.Ignore
+}
+
+// Rules implements Webhook interface
+func (s *SubscriptionWebhook) Rules() []admissionregv1.RuleWithOperations { return rules }
+
+// GetURI implements Webhook interface
+func (s *SubscriptionWebhook) GetURI() string { return "/subscription-validation" }
+
+// SideEffects implements Webhook interface
+func (s *SubscriptionWebhook) SideEffects() admissionregv1.SideEffectClass {
+	return admissionregv1.SideEffectClassNone
+}
+
+// Validate is the incoming request even valid?
+func (s *SubscriptionWebhook) Validate(req admissionctl.Request) bool {
+	valid := true
+	valid = valid && (req.UserInfo.Username != "")
+	valid = valid && (req.Kind.Kind == "Subscription")
+
+	return valid
+}
+
+// isLogging45Request is a helper to consolidate logic. Returns true
+// if the request is for the cluster-logging or elasticsearch-operator
+// subscription on 4.5 channel
+func (s *SubscriptionWebhook) isLogging45Request(subscriptionReq *subscriptionRequest) bool {
+	if subscriptionReq.Spec.Name == LoggingSubName || subscriptionReq.Spec.Name == ESSubName {
+		if subscriptionReq.Spec.Channel == Channel {
+			return true
+		}
+	}
```

Comment:
```suggestion
	if subscriptionReq.Spec.Channel == Channel && (subscriptionReq.Spec.Name == LoggingSubName || subscriptionReq.Spec.Name == ESSubName) {
		return true
	}
```

This could work a little better? Shortcutting on the channel check at first.

Reply:
Sure :) my concern was readability, but I can add comments